### ISSUE TITLE
error while testing triplet loss trained nework

### ISSUE THREAD

**Issue by AkankshaJ** [cmusatyalab/openface#111]
Hi,

I have trained one model using triplet loss. My input image size is 3 x 128 x 128. Now when I am using the same size image and do net:forward(img) , I get the following view error; 

install/share/lua/5.1/torch/Tensor.lua:462: Wrong size for view. Input size: 200x1024x2x2. Output size: 200x1024
 * 200 is batch size

Now when i am using the same size image ( on which my model is trained ) to test the network , why I am getting this error?? Is there any issue when the network is decoded? My model is

 net:add(nn.SpatialConvolutionMM(3, 64, 5, 5)) 
   net:add(nn.SpatialBatchNormalization(64))
   net:add(nn.ReLU())

 
  net:add(nn.SpatialMaxPooling(3, 3, 2, 2, 1, 1)) 

   -- Inception (2)
   net:add(nn.SpatialConvolutionMM(64, 64, 1, 1))
   net:add(nn.SpatialBatchNormalization(64))
   net:add(nn.ReLU())
   net:add(nn.SpatialConvolutionMM(64, 192, 3, 3))
   net:add(nn.SpatialBatchNormalization(192))
   net:add(nn.ReLU())

  net:add(nn.SpatialMaxPooling(3, 3, 2, 2, 1, 1))

   -- Inception (3a)
   net:add(nn.Inception{
     inputSize = 192,
     kernelSize = {3, 5},
     kernelStride = {1, 1},
     outputSize = {128, 32},
     reduceSize = {96, 16, 32, 64},
     pool = nn.SpatialMaxPooling(3, 3, 2, 2),
     batchNorm = true
   })

   -- Inception (3b)
   net:add(nn.Inception{
     inputSize = 256,
     kernelSize = {3, 3},
     kernelStride = {1, 1},
     outputSize = {128, 64},
     reduceSize = {96, 32, 64, 64},
     pool = nn.SpatialLPPooling(256, 2, 3, 3),
     batchNorm = true
   })

   -- Inception (3c)
   net:add(nn.Inception{
     inputSize = 320,
     kernelSize = {3, 2},
     kernelStride = {2, 2}, --15
     outputSize = {256, 64},
     reduceSize = {128, 32, nil, nil},
     pool = nn.SpatialMaxPooling(3, 3, 2, 2),
     batchNorm = true
   })

   -- Inception (4a)
   net:add(nn.Inception{
     inputSize = 640,
     kernelSize = {3, 3},
     kernelStride = {1, 1},
     outputSize = {192, 64},
     reduceSize = {96, 32, 128, 256},
     pool = nn.SpatialLPPooling(640, 2, 3, 3),
     batchNorm = true
   })
   -- Inception (4b)
   net:add(nn.Inception{
     inputSize = 640,
     kernelSize = {3, 2},
     kernelStride = {2, 2}, --8
     outputSize = {256, 128},
     reduceSize = {160, 64, nil, nil},
     pool = nn.SpatialMaxPooling(3, 3, 2, 2),
     batchNorm = true
   })

   -- Inception (5a)
   net:add(nn.Inception{
              inputSize = 1024,
              kernelSize = {3,2},
              kernelStride = {1,1},
              outputSize = {384,128},
              reduceSize = {192, 128, 384,128},
              pool = nn.SpatialLPPooling(896, 2, 3, 3),
              batchNorm = true
   })

   -- Inception (5b)
   net:add(nn.Inception{
              inputSize = 1024,
              kernelSize = {3,2},
              kernelStride = {2,2}, --4
              outputSize = {384,128},
              reduceSize = {192, 128, 384,128},
              pool = nn.SpatialMaxPooling(3, 3, 2, 2),
              batchNorm = true
   })

   net:add(nn.SpatialAveragePooling(4,4,4,4))

   net:add(nn.Reshape(1024))

   net:add(nn.View(1024))
   net:add(nn.Linear(1024, 256))
   net:add(nn.Normalize(2))

Thanks a lot !

**Comment by bamos** [comment]
Hi @AkankshaJ - why did you delete our issue template? The issue tracker is for specific bug reporting and development discussions, in the future please post questions like this to our mailing list.

The output of your pooling layer isn't 1024-dimensional when you're testing and I'm not sure what could be causing this other than using a differently sized input to the network. As a debugging strategy you can add more `nn.View` layers throughout the network to make sure the sizes are as you expect.

-Brandon.
